### PR TITLE
Search package: add i18n support for auto added search block

### PR DESCRIPTION
**Pull request by kangzj**
#### Changes proposed in this Pull Request:
The PR adds i18n support for auto added Search input, and is a followup for #22613.

#### Jetpack product discussion
https://github.com/Automattic/jetpack/pull/22613#discussion_r809930841

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a site with Search subscription
* Change theme to a FSE theme (e.g. 2020), remove search input in the footer if any
* Change the language of the site to non-English
* Update translations if any (Dashboard -> Updates)
* Run `jetpack docker wp jetpack-search auto_config wordpress` <- replace wordpress to a local user name or ID
* Ensure Search input is added in footer, label and button text translated

image

### CODE REVIEW

File: projects/packages/search/src/instant-search/class-instant-search.php [hunk 1]
```diff
@@ -620,7 +620,11 @@ protected function template_parts_have_search_block() {
 	 * @param {string} $block_content - the content to append the search block.
 	 */
 	public static function inject_search_widget_to_block( $block_content ) {
-		$search_block = '<!-- wp:search {"label":"Jetpack Search","buttonText":"Search"} /-->';
+		$search_block = sprintf(
+			'<!-- wp:search {"label":"%s","buttonText":"%s"} /-->',
```
Comment (jsnmoon) [inline projects/packages/search/src/instant-search/class-instant-search.php:624]: Why did we change from "Jetpack Search" to "Search"? Also, what are your thoughts on omitting the label altogether?

**Review by jsnmoon — APPROVED** [review]
Works as described! Left a small note, but looks okay to ship as-is.

**Review by jsnmoon — CHANGES_REQUESTED** [review]
Looks like our tests need updating, actually 😬

**Comment by kangzj** [comment]
Thanks for the speedy review. Glad the unit tests worked haha 😄 

Removed label text and fixed tests in 686f1aa2c8

File: projects/packages/search/src/instant-search/class-instant-search.php (hunk 1 above)
Comment (kangzj) [reply projects/packages/search/src/instant-search/class-instant-search.php]: I like the idea to remove the label, the purpose of the block is very obvious.

**Review by jsnmoon — APPROVED** [review]
:shipit:

**Pull request merged**